### DISCUSSION TITLE
PureHDF.SourceGenerator - Bindigns

### DISCUSSION THREAD

**Discussion by frsgoncal** [Apollo3zehn/PureHDF#39] [General]
Hello, I'm trying to use your lob PureHDF.SourceGenerator, to use binding to my hdf5 file, but for some reason when I try to do:  `var bindings = new MyGeneratedH5Bindings(h5File);` that say the MyGeneratedH5Bindings does not contain a constructor that takes 1 arg. The steps I do: Instal dotnet lib -> dotnet restore -> lib is on dependencies -> added `[H5SourceGenerator(filePath: MY_FILE)]
    internal partial class MyGeneratedH5Bindings { };` to my code. Can you help me?

**Comment by Apollo3zehn** [comment]
Did you also add the `PureHDF.SourceGenerator` package to your list of package references? When I tried to reproduce your problem I accidentially did only install `PureHDF` and then I got your error. After installing the correct package and restarting VS Code (the Microsoft source generators IDE support still seem to be a bit buggy), it worked for me:

![image](https://github.com/Apollo3zehn/PureHDF/assets/20972129/f6ac560f-520c-4a15-97b6-3137dba3db33)


![image](https://github.com/Apollo3zehn/PureHDF/assets/20972129/fc0f9f78-2d5e-4225-86d4-06813ed75b82)

**Reply by frsgoncal** [reply]
![image](https://github.com/Apollo3zehn/PureHDF/assets/82593913/64e03edb-86e5-46f0-a6db-7c6f08603795)
I have it included and can read the h5 file normaly with .OpenRead and also use with .Dataset and I get all info I need, but every time the MyGeneratedH5Bindings tell me that not contain constructor with 1 arg. 
![image](https://github.com/Apollo3zehn/PureHDF/assets/82593913/ecf43614-9a17-4366-b758-820889efd751)
I'm using VS Pro, maybe just work for VS Code??!?!

**Reply by Apollo3zehn** [reply]
It should also work with Visual Studio, I will install and test it. I come back to you later when I am done :-) (later today)

**Reply by frsgoncal** [reply]
Forget the last message, already work, thanks a lot! I think is Because I have put [H5SourceGenerator(filePath: h5test.FILE_PATH)]
    internal partial class MyGeneratedH5Bindings { }; in wrong possition!
    Keep your great job!

**Reply by Apollo3zehn** [reply]
perfect :-)

**Comment by Apollo3zehn** [comment]
Another reason I can think of is if your file cannot be read correctly but I have tons of unit tests so that is unlikely but not impossible (mainly because I am unable to create old style HDF5 files so I can only test 95 % of PureHDF's code).
